docs/user-guide/rug/conventions.md

```diff
@@ -101,288 +70,262 @@ the [`CHANGELOG.md` format][changelog].
-for the corresponding parameter.  In addition, you should provide a
-meaningful value for the `@validInput` parameter annotation so, when
-people provide an invalid value, they are provided with a useful error
-message.
-
-#### Mandatory and Optional Parameters
-
-If a parameter is optionally provided on invocation, using the
-`@optional` annotation, then it is recommended to supply a valid
-`@default` annotation also so that there is some predictability of how
-the Rug will function if no parameter is supplied.
-
-### TypeScript Decorators
-
-There are TypeScript equivalents to the Rug DSL annotations described above, and
-they follow the same conventions:
-
-```TypeScript
-...
-@Editor("Simple", "My simple editor")
-@Tags("simple", "decorators")
-class SimpleEditor {
-    @Parameter({description: "The name", pattern: "^.*$$"})
-    name: string = "Not reserved"
-...
-```
-The `@Editir` decorator labels the class as an `editor`, with `name` and `description`
-as its two parameters respectively. `@Generator` and `@Reviewer` are the TypeScript
-decorator equivalents of the `generator` and `reviewer` DSL keywords. The `@Tags` decorator takes a variable number of strings and `@Parameter` is used to declare parameters. TypeScript parameter declarations support the equivalent fields as Rug DSL and conform to the following
-interface:
-
-```TypeScript
-interface Parameter {
-  pattern: string
-  required?: boolean
-  description?: string
-  displayName?: string
-  validInput?: string
-  displayable?: boolean
-  maxLength?: number
-  minLength?: number
-  tags?: string[]
+Here is a representative example of a Rug implemented in TypeScript following the above points:
+
+```typescript
+import { EditProject } from "@atomist/rug/operations/ProjectEditor";
+import { Project } from "@atomist/rug/model/Project";
+import { Pattern } from "@atomist/rug/operations/RugOperation";
+import { Editor, Parameter, Tags } from "@atomist/rug/operations/Decorators";
+
+@Editor("MyFirstEditor", "A sample Rug TypeScript editor to start playing with.")
+@Tags("documentation")
+export class MyFirstEditor implements EditProject {
+
+    @Parameter({
+        displayName: "Some Input",
+        description: "example of how to specify a parameter using decorators",
+        pattern: Pattern.any,
+        validInput: "a description of the valid input",
+        minLength: 1,
+        maxLength: 100
+    })
+    input_parameter: string;
+
+    edit(project: Project) {
+        project.addFile("hello.txt", "Hello, World!\n" + this.input_parameter + "\n");
+    }
 }
+
+export const myFirstEditor = new MyFirstEditor();
 ```
-The name and default value of a parameter are taken from the code itself.
 
-### Rug DSL: `let` Declarations
+!!! tip
+    We do not issue specific conventions regarding TypeScript code but it is certainly a good idea to rely on a [TypeScript linter][tslint] to catch common pitfalls in your code.
```

Comment:
Perhaps we should move this out of a tip and say that it is convention to use tslint and tsfmt using the defaults?